synapse/federation/transport/server.py

```diff
@@ -1474,18 +1512,24 @@ async def on_GET(self, origin, content, query, room_id):
 )
 
 
-def register_servlets(hs, resource, authenticator, ratelimiter, servlet_groups=None):
+def register_servlets(
+    hs: "synapse.server.HomeServer",
```

Comment:
Out of curiosity I always use `from synapse.app.homeserver import HomeServer` for these type hints. Should we be using this instead? (Also any reason for using the full dotted module name here instead of a `from` import? We don't seem to do that often.)

Reply:
> Out of curiosity I always use from synapse.app.homeserver import HomeServer

There isn't a `HomeServer` in `synapse.app.homeserver` (except by import from `synapse.server`. Do you mean `SynapseHomeServer`? In which case, that won't really be accurate for a worker process.

> (Also any reason for using the full dotted module name here instead of a from import? We don't seem to do that often.)

Fair; I'll change this.

Reply:
I suspect I've made a bunch of imports in correct! Oops. I'll put up a PR to fix this.